packages/docs-website/docs/protocol-docs/client-specification/json-rpc-api.md

```diff
@@ -0,0 +1,420 @@
+| allocations             | `Allocation[]`    | Array of funding amounts for participants  |
+| appData                 | string            | Encoded initial state of app               |
+| appDefinition           | string            | Address of contract governing the channel  |
+| channelId               | string            | Unique channel identifier                  |
+| status                  | `ChannelStatus`   | Current status of channel                  |
+| turnNum                 | number            | Channel nonce                              |
+| challengeExpirationTime | number (optional) | Time current challenge on channel elapses  |
+
+### Errors
+
+TODO: (HIGH) Clean up error message types and codes
+
+| Code | Message            | Description                                 |
+| ---- | ------------------ | ------------------------------------------- |
+| 400  | Channel not found  | Could not find channel to update in storage |
+| 401  | Invalid transition | Illegal state transition proposed           |
+| 402  | Invalid app data   | Incorrect encoded app information           |
+| 403  | Not your turn      | Cannot update channel                       |
+| 404  | Channel closed     | Channel no longer accepting updates         |
+
+## JoinChannel
+
+Called when you would are joining a channel that has been created. Generally, creating a channel is done, and a notification is sent to the desired counterparty, who then dispatches the `JoinChannel` request.
+
+### Parameters
+
+| Name      | Type      | Description               |
+| --------- | --------- | ------------------------- |
+| channelId | `Bytes32` | Unique channel identifier |
+
+### Response
+
+| Name                    | Type              | Description                                |
+| ----------------------- | ----------------- | ------------------------------------------ |
+| participants            | `Participant[]`   | Identifying information members of channel |
+| allocations             | `Allocation[]`    | Array of funding amounts for participants  |
+| appData                 | string            | Encoded initial state of app               |
+| appDefinition           | string            | Address of contract governing the channel  |
+| channelId               | string            | Unique channel identifier                  |
+| status                  | `ChannelStatus`   | Current status of channel                  |
+| turnNum                 | number            | Channel nonce                              |
+| challengeExpirationTime | number (optional) | Time current challenge on channel elapses  |
+
+### Errors
+
+| Code | Message            | Description                                 |
+| ---- | ------------------ | ------------------------------------------- |
+| 1100 | Channel not found  | Could not find channel to update in storage |
+| 1101 | Invalid transition | Channel cannot be joined                    |
+
+## CloseAndWithdraw
```

Comment:
This is also probably the wrong API and should be split up into `close`, `defund`, and `withdraw`